docs/proposals/nginx-extensions.md

```diff
@@ -0,0 +1,941 @@
+
+#### Future Work
+
+- Add support for remote authentication or other authentication strategies.
+
+#### Alternatives
+
+- Begin with a Filter: Rather than beginning with an Inherited Policy, which is complex, we could start by adding an authentication Filter. This would allow us to roll out an authentication solution quickly to support the core use case for Application Developers. The use case for Cluster Operators applying authentication Policies is less well-known than the Application Developer's use case. Instead of leading with the Cluster Operator use case, we can begin with a Filter and wait for user feedback. Later, we can add a Policy and define how the Filter and Policy interact.
+- Wait for Gateway API GEP on authentication: There is an open [GEP issue](https://github.com/kubernetes-sigs/gateway-api/issues/1494) for authentication in the Gateway API repository. Instead of rolling out our own authentication solution, we could wait or even champion this GEP. The advantages of waiting are that if the GEP is implemented, we won't have to deprecate our bespoke authentication Policy in favor of the GEP's implementation, and we won't have to maintain an additional Policy. However, we don't know how long it will take the GEP to move from open to implementable. Authentication is a critical feature, and it may not be prudent to wait.
+
+### Observability
+
+_Extension type:_ Direct Policy
+
+_Resource type:_ CRD
+
+_Role(s):_ Application Developer
+
+_Extension point:_ HTTPRoute, HTTPRoute rule
+
+_NGINX context(s):_ http, server, location
+
+Features:
+
+- OTEL Tracing: enable/disable by Route or routing rule. Set span name, attributes, and context.
+
+NGINX directives:
+
+- [`otel_trace`](https://nginx.org/en/docs/ngx_otel_module.html#otel_trace)
+- [`otel_trace_context`](https://nginx.org/en/docs/ngx_otel_module.html#otel_trace_context)
+- [`otel_span_name`](https://nginx.org/en/docs/ngx_otel_module.html#otel_span_name)
+- [`otel_span_attr`](https://nginx.org/en/docs/ngx_otel_module.html#otel_span_attr)
+
+#### Future Work
+
+Add support for:
+
+- OTEL logging
+- Metrics
+
+#### Alternatives
+
+- Combine with OTEL settings in Gateway Settings for one OTEL Policy: Rather than splitting tracing across two Policies, we could create a single tracing Policy. The issue with this approach is that some tracing settings -- such as exporter endpoint -- should be restricted to Cluster Operators, while settings like attributes should be available to Application Developers. If we combine these settings, RBAC will not be sufficient to restrict access across the settings. We will have to disallow certain fields based on the resource the Policy is attached to. This is a bad user experience.
+- Inherited Policy: An Inherited Policy would be useful if there is a use case for the Cluster Operator enforcing or defaulting the OTEL tracing settings included in this policy.
```

Comment:
> Question, if it's an inherited policy, does a parent policy HAVE to exist, or is that optional?
I don't see anything in the GEP that suggests a parent policy is required.

Because of the complexities around Inherited Policies, I think we should be very sure that a use case for defaults and overrides exist. 

In this case, we need to figure out if there's a use case for the ClusterOperator to provide sane defaults or enforce the following OTEL settings:

- [`otel_trace_context`](https://nginx.org/en/docs/ngx_otel_module.html#otel_trace_context)
- [`otel_span_name`](https://nginx.org/en/docs/ngx_otel_module.html#otel_span_name)
- [`otel_span_attr`](https://nginx.org/en/docs/ngx_otel_module.html#otel_span_attr)

The [`otel_trace`](https://nginx.org/en/docs/ngx_otel_module.html#otel_trace) setting is different since it will also be exposed in the Gateway Settings. So the ClusterOperator will be able to enable/disable at the Gateway level.

Reply:
To clarify, I don't mean "is there a use case for a parent policy", but more literally, does a parent policy have to be created in a cluster first before a lower level policy can exist?

Reply:
Sorry, I answered that question but I messed up the formatting 🤦‍♀️

Reply:
I edited the original response, but I'll restate here:

I don't see anything in the GEP that suggests a parent policy is required.